examples/asr/asr_chunked_inference/rnnt/speech_to_text_buffered_infer_rnnt.py

```diff
@@ -0,0 +1,282 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Script to perform buffered inference using RNNT models.
+
+# Middle Token merge algorithm
```

Comment:
> Is there any reference for these two methods? or can we explain the idea generally?
No, they are developed here. The middle token algorithm is just general enough that there may be a reference somewhere. The LCS algo will have a corresponding notebook detailing what it is since there's no reference.

> Other option can be have a section on buffering/streaming in the doc which explain what they do and when they should be used.
Ill add a section for buffered inference for transducers in the future (hands full right now). @jbalam-nv id like you to write a subsection there for CTC.